server/schedulers/hot_region.go

```diff
@@ -93,73 +104,52 @@ type hotScheduler struct {
 	sync.RWMutex
 	leaderLimit uint64
 	peerLimit   uint64
-	types       []BalanceType
+	types       []rwType
+	r           *rand.Rand
+
+	// states across multiple `Schedule` calls
+	readPendings   map[*pendingInfluence]struct{}
+	writePendings  map[*pendingInfluence]struct{}
+	regionPendings map[uint64][2]*pendingInfluence
 
-	// store id -> hot regions statistics as the role of leader
-	stats           *storeStatistics
-	r               *rand.Rand
-	readPendings    map[*pendingInfluence]struct{}
-	writePendings   map[*pendingInfluence]struct{}
+	// temporary states but exported to API or metrics
+	stats           *storeStatistics // store id -> hot regions statistics
 	readPendingSum  map[uint64]Influence
 	writePendingSum map[uint64]Influence
 	readScores      *ScoreInfos
 	writeScores     *ScoreInfos
-	pendingOpInfos  map[uint64]*list.List
 }
 
 func newHotScheduler(opController *schedule.OperatorController) *hotScheduler {
 	base := NewBaseScheduler(opController)
-	pendingOpInfos := make(map[uint64]*list.List)
 	return &hotScheduler{
 		name:           HotRegionName,
 		BaseScheduler:  base,
 		leaderLimit:    1,
 		peerLimit:      1,
 		stats:          newStoreStatistics(),
-		types:          []BalanceType{hotWrite, hotRead},
+		types:          []rwType{write, read},
 		r:              rand.New(rand.NewSource(time.Now().UnixNano())),
 		readScores:     NewScoreInfos(),
 		writeScores:    NewScoreInfos(),
 		readPendings:   map[*pendingInfluence]struct{}{},
 		writePendings:  map[*pendingInfluence]struct{}{},
-		pendingOpInfos: pendingOpInfos,
+		regionPendings: make(map[uint64][2]*pendingInfluence),
 	}
 }
 
 func newHotReadScheduler(opController *schedule.OperatorController) *hotScheduler {
-	base := NewBaseScheduler(opController)
-	pendingOpInfos := make(map[uint64]*list.List)
-	return &hotScheduler{
-		BaseScheduler:  base,
-		leaderLimit:    1,
-		peerLimit:      1,
-		stats:          newStoreStatistics(),
-		types:          []BalanceType{hotRead},
-		r:              rand.New(rand.NewSource(time.Now().UnixNano())),
-		readScores:     NewScoreInfos(),
-		writeScores:    NewScoreInfos(),
-		readPendings:   map[*pendingInfluence]struct{}{},
-		writePendings:  map[*pendingInfluence]struct{}{},
-		pendingOpInfos: pendingOpInfos,
-	}
+	ret := newHotScheduler(opController)
+	ret.name = ""
```

Comment:
why reset name?

Reply:
The original code didn't set `name`, so it should be empty string.